### ISSUE TITLE
Setting to Disable .js / .ts Autocomplete?

### ISSUE THREAD

**Issue by karlhorky** [vunguyentuan/vscode-css-variables#58]
Hi @vunguyentuan 👋 hope you are well :)

Would you consider adding a setting to disable [the `.js` / `.ts` autocomplete feature](https://github.com/vunguyentuan/vscode-css-variables/issues/4) for projects that do not use CSS-in-JS?

Currently, in a project that does not use CSS-in-JS, the extension fills the autocomplete entries all over the place in JS, JSX, TS and TSX files with useless entries for CSS variables:

<img width="541" alt="Screenshot 2023-01-24 at 11 40 59" src="https://user-images.githubusercontent.com/1935696/214272831-f981cbfc-9099-4d4b-943b-f4903bdb83d7.png">

For example, adding a new setting called `cssVariables.autocompleteInLanguages`, default values below:

```json
{
  "cssVariables.autocompleteInLanguages": [
    "css",
    "less",
    "scss",
    "postcss",
    "javascript",
    "javascriptreact",
    "typescriptreact"
  ],
}
```

Or, if this is based on file extension instead of language in the file, a new setting called `cssVariables.autocompleteInFiles`, default values:

```json
{
  "cssVariables.autocompleteInFiles": [
    "*.css",
    "*.scss",
    "*.sass",
    "*.less",
    "*.js",
    "*.jsx",
    "*.ts",
    "*.tsx"
  ]
}
```

**Comment by Cellule** [comment]
:+1: I do not use styles in javascript and the autocomplete is just pollution. I really only want them in css files.
It's fine if that's the default, but I was expecting a setting

**Comment by karlhorky** [comment]
@Cellule @vunguyentuan thanks for the setting!

**Comment by karlhorky** [comment]
@vunguyentuan is this new `cssVariables.languages` setting working completely?

I'm still receiving suggestings in a `.ts` files (`typescript` syntax), even though I have disabled those:

![Screenshot 2024-02-21 at 20 34 29](https://github.com/vunguyentuan/vscode-css-variables/assets/1935696/47ec9fc9-f716-4145-a83a-a96852d63b66)

Disabling the extensions makes these suggestions disappear.

This is my configuration here:

```js
{
  "cssVariables.languages": [
    // "astro",
    // "svelte",
    // "vue",
    // "vue-html",
    // "vue-postcss",
    "scss",
    "postcss",
    "less",
    "css",
    "html",
    // "javascript",
    // "javascriptreact",
    // "typescript",
    // "typescriptreact",
    "source.css.styled"
  ]
}
```

**Comment by Cellule** [comment]
I admit something is off. It worked fine initially with my fork, then once it was published it no longer seemed to work
Haven't had time yet to investigate

**Comment by karlhorky** [comment]
@vunguyentuan can we reopen this issue?

**Comment by Cellule** [comment]
@karlhorky I found my bug 😅
https://github.com/vunguyentuan/vscode-css-variables/pull/85
Sorry about that, hopefully it gets merged and released soon :)

**Comment by karlhorky** [comment]
Amazing, thanks! Watching #85 now :)

**Comment by karlhorky** [comment]
@vunguyentuan @Cellule thanks for the PR and merge!

I can confirm that this is fixed in the new `2.7.1` version which was just published 🙌
